packages/pc-external-ready.nix

```
@@ -0,0 +1,30 @@
+{
+  writeShellApplication,
+  curl,
+  jq,
+  # Process compose instance
+  PC_PORT_NUM ? 32017,
+  # Process to wait for
+  TARGET_PROCESS_NAME ? "postgres",
```

we may want to force caller to specify these. `"postgres"` is also a _really_ weird default
```suggestion
  PC_PORT_NUM,
  # Process to wait for
  TARGET_PROCESS_NAME,
```

---

alternatively, create an argument parser for both of these arguments instead of putting them in the derivation. that may be cleaner.

---

1. can export it as lib overlay (i was thinking it initially). it will named like n1.lib.mkPcReady

2. do option parsing

3. use native wait in updated pc https://github.com/F1bonacc1/process-compose/pull/329 (they open to add timeout too)

---

i think use native wait with external timeout. 


timeout 20s pc is-ready --wait

will remove nix package